v2/AdminCore.WebApp/Controllers/HolidayController.cs

```diff
@@ -22,74 +27,184 @@ namespace AdminCore.WebApi.Controllers
+      _employeeService = employeeService;
       _mapper = mapper;
     }
 
     [HttpGet]
     public IActionResult GetAllHolidays()
     {
-      return Ok();
+      var holidaysDtos = _holidayEventService.GetByType(EventTypes.AnnualLeave);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+      return Ok(holidays);
     }
 
     [HttpGet("{holidayId}")]
     public IActionResult GetHolidayByHolidayId(int holidayId)
     {
-      return Ok();
+      var holiday = _holidayEventService.GetEvent(holidayId);
+      if (holiday.Status == MessageConstants.MsgStatusSuccess)
+      {
+        return Ok(_mapper.Map<HolidayViewModel>(holiday.Payload));
+      }
+
+      return Ok(holiday.Payload);
     }
 
     [HttpGet("findByEmployeeId/{employeeId}")]
     public IActionResult GetHolidayByEmployeeId(int employeeId)
     {
-      return Ok();
+      ResponseMessage<IList<EventDto>> holidaysDtos = _holidayEventService.GetEventsByEmployeeId(employeeId);
+      var holidays = _mapper.Map<ResponseMessage<IList<EventDto>>, List<HolidayViewModel>>(holidaysDtos);
+      return Ok(holidays);
     }
 
     [HttpPost]
     public IActionResult CreateHoliday(CreateHolidayViewModel viewModel)
     {
-      return Ok();
+      IList<string> errors = ValidateCreateHolidayViewModel(viewModel);
+      return CreateResponseBasedOnErrors(errors, viewModel);
     }
 
     [HttpPut]
-    public IActionResult UpdateHoliday(UpdateHolidayViewModel viewModel)
+    public IActionResult UpdateHoliday(UpdateHolidayViewModel updateHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<UpdateHolidayViewModel, EventDto>(updateHoliday);
+      return Ok(_holidayEventService.UpdateEvent(eventDto));
     }
 
     [HttpPut("approveHoliday")]
     public IActionResult ApproveHoliday(ApproveHolidayViewModel approveHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<ApproveHolidayViewModel, EventDto>(approveHoliday);
+      return Ok(_holidayEventService.ApproveEvent(eventDto));
     }
 
     [HttpPut("cancelHoliday")]
     public IActionResult CancelHoliday(CancelHolidayViewModel cancelHoliday)
     {
-      return Ok();
+      var eventDto = _mapper.Map<CancelHolidayViewModel, EventDto>(cancelHoliday);
+      return Ok(_holidayEventService.CancelEvent(eventDto));
     }
 
     [HttpPut("rejectHoliday")]
     public IActionResult RejectHoliday(RejectHolidayViewModel rejectHoliday)
     {
-      return Ok();
+      var responses = _holidayEventService.RejectEvent(
+        rejectHoliday.EventId,
+        rejectHoliday.Message,
+        int.Parse(_authenticatedUser.RetrieveUserId()));
+      return Ok(responses);
     }
 
-    [HttpGet("findByDateBetween/{eventDate}")]
-    public IActionResult GetHolidayByDateBetween(EventDate eventDate)
+    [HttpGet("findByDateBetween/{startDate}/{endDate}")]
+    public IActionResult GetHolidayByDateBetween(string startDate, string endDate)
```

Comment:
You guys should look at Parameter Binding: https://docs.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/parameter-binding-in-aspnet-web-api